### PR TITLE
Fixes [any] operator handling in built-in aggregators for null/empty collections/arrays

### LINKED ISSUE #10126: Fast Aggregations on [any] operator on empty array does not work
For  builtin aggregations like Count, LongSum etc.
If we used them with an `attributePath` on list/array field like;

```
class Person implements Portable {
...
    int[] valueArray;
...
}
```

if we initialize valueArray with an empty array`valueArray = new int[]{};`

```
map.aggregate(Aggregators.<Map.Entry<Integer, Person>>integerSum.integerSum("valueArray[any]"));
```
throws NullPointerException instead of returning 0, 

```
map.aggregate(Aggregators.<Map.Entry<Integer, Person>>count("valueArray[any]"));
```
returns 1 instead of 0.

Other builtin aggregators may also fail, they should be tested.

### PR DESCRIPTION
**Pull request by tombujok**
Fixes #10126

Allows differentiating whether a null in the MultiResult is a null value because the value of the field was null or whether the null means that the targetObject was null.

For query evaluation the difference is not important - and we need to return null in both cases there. (see [any] corner cases in the reference manual)
For aggregations it makes a difference and we need this context knowledge there.

MultiResult is a structure returned if an extraction contains an `[any]` operator, since a single read of a `person.hands[any].length` expression may return more than one result.

Here's more info about `[any]`: http://docs.hazelcast.org/docs/3.8/manual/html-single/index.html#querying-in-collections-and-arrays

### CODE REVIEW

**Comment by tombujok** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java [hunk 1]
```diff
@@ -69,6 +69,7 @@
 public class MultiResult<T> {
 
     private List<T> results;
+    private boolean targetNullOrEmpty;
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java]: an explanatory comment can be added to code.

**Comment by ahmetmircik** [comment]
is this pr backportable?

File: hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java (hunk 1 above)
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java]: Ok, sure. I'll add it.

**Comment by tombujok** [comment]
@ahmetmircik yep, it's back-portable.

File: hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java (hunk 1 above)
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java]: Added

**Comment by tombujok** [comment]
Do not merge for now. I still need to verify sth here.

**Comment by tombujok** [comment]
Two issues before this can get merged:
- it depends on this https://github.com/hazelcast/hazelcast/pull/10143
- there is still no portable support

File: hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java [hunk 2]
```diff
@@ -70,6 +70,8 @@
 
     private List<T> results;
 
+    private boolean nullOrEmptyTarget;
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java]: can we have this kind of comment back? https://github.com/hazelcast/hazelcast/pull/10131/commits/9ded149c1085d829813626046f47206b8b375522#diff-08181c4c9677f20321b81b482ff1d796R75

File: hazelcast/src/main/java/com/hazelcast/query/impl/getters/GetterFactory.java [hunk 3]
```diff
@@ -47,6 +49,9 @@ public static Getter newFieldGetter(Object object, Getter parentGetter, Field fi
                 returnType = getCollectionType(collection);
             }
             if (returnType == null) {
+                if (modifierSuffix.equals(ANY_POSTFIX)) {
+                    return NullMultiValueGetter.NULL_MULTIVALUE_GETTER;
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/query/impl/getters/GetterFactory.java]: `NullMultiValueGetter.INSTANCE`?
 or can be imported statically, current form can make reads hard

File: hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java (hunk 2 above)
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java]: hmm weird, I didn't delete it on purpose, somehow it disappeared... Sure I will add it.

File: hazelcast/src/main/java/com/hazelcast/query/impl/getters/GetterFactory.java (hunk 3 above)
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/query/impl/getters/GetterFactory.java]: Done.

File: hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java (hunk 2 above)
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/query/impl/getters/MultiResult.java]: Done.

**Comment by tombujok** [comment]
run-lab-run

**Pull request merged**